src/thor/route_action.cc

```diff
@@ -152,6 +173,15 @@ void thor_worker_t::route(Api& request) {
 thor::PathAlgorithm* thor_worker_t::get_path_algorithm(const std::string& routetype,
                                                        const valhalla::Location& origin,
                                                        const valhalla::Location& destination) {
+  // Tell all the algorithms where to track the expansion
+  // If the pointer has been reset it will have no effect
+  multi_modal_astar.set_expansion(expansion_);
+  timedep_forward.set_expansion(expansion_);
+  timedep_reverse.set_expansion(expansion_);
+  astar.set_expansion(expansion_);
+  bidir_astar.set_expansion(expansion_);
```

Comment:
the algorithms get a weak pointer to this geojson object so they can tell that if it isnt something they should be tracking if its not backed by a real shared pointer